### PR TITLE
Implement Delete account possibility

### PR DESCRIPTION
**Pull request by VyacheslavDudin**
Create backend flow for deleting personal account from the application.
Use access token to send DELETE request.

User Story: As an authorized user, I want to be able to delete my account, so that my personal data would be removed if I want to stop using the application.
Acceptance Criteria:
1. The Delete account option is represented in the form of a button.
2. The text in the button is "Delete Account".
3. The button's colour is red, the text's colour is white.
4. The delete option is located in the Edit profile form.
5. The option will delete the registered user's account permanently.
6. The option will delete the user's data.

### CODE REVIEW

File: backend/src/data/repositories/userRepository.ts [hunk 1]
```diff
@@ -20,14 +19,21 @@ class UserRepository extends Repository<User> {
     return this.findOne(id);
   }
 
+  async deleteUser(id: string): Promise<User> {
```
Comment (AlikRakhmonov) [inline backend/src/data/repositories/userRepository.ts:22]: delete doesn't return anything (void)

File: backend/src/services/userService.ts [hunk 2]
```diff
@@ -13,6 +13,13 @@ export const getUserById = async (id: string) => {
   return user;
 };
 
+export const deleteUser = async (id: string) => {
+  const deletedUser = await getCustomRepository(UserRepository).deleteUser(id);
+  return {
+    user: fromUserToUserClient(deletedUser)
```
Comment (AlikRakhmonov) [inline backend/src/services/userService.ts:19]: incorrect, see 1 comment

File: frontend/src/common/models/user/IUser.ts [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
 }
+
+interface IUserState {
+  data: IUser | null;
+  isLoading: boolean;
+  isAuthorized: boolean;
+}
+
+export type { IUser, IUserState };
+
```
Comment (vlvereta) [inline frontend/src/common/models/user/IUser.ts:17]: 1 file = 1 interface
**IUserState** is laying inside appropriate reducer file

File: frontend/src/common/models/user/IUser.ts [hunk 4]
```diff
@@ -1,8 +1,17 @@
-export interface IUser {
+interface IUser {
```
Comment (vlvereta) [inline frontend/src/common/models/user/IUser.ts:1]:
```suggestion
export interface IUser {
```

File: frontend/src/containers/EditProfile/styles.module.sass [hunk 5]
```diff
@@ -10,3 +10,9 @@
 .imageEditButton
   margin: 5px 0
   min-width: 150px
+
+.deleteAccountButton
+  margin: 5px 0
+  min-width: 150px
+  background-color: red
```
Comment (vlvereta) [inline frontend/src/containers/EditProfile/styles.module.sass:17]: colors should be in global colors

File: frontend/src/sagas/user.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ import {
   fetchUserRoutine,
   editProfileRoutine,
   addNewUserRoutine,
+  deleteAccountRoutine as delAccount,
```
Comment (vlvereta) [inline frontend/src/sagas/user.ts:6]: Leave it as **deleteAccountRoutine** like the rest.

File: frontend/src/containers/EditProfile/styles.module.sass [hunk 7]
```diff
@@ -10,3 +10,9 @@
 .imageEditButton
   margin: 5px 0
   min-width: 150px
+
+.deleteAccountButton
+  margin: 5px 0
+  min-width: 150px
+  background-color: var(--del-btn-background)
+  color: white
```
Comment (vlvereta) [inline frontend/src/containers/EditProfile/styles.module.sass:18]: This one?

**Pull request merged**